pkg/test/framework/components/echo/common/deployment/echos.go

```diff
@@ -75,18 +75,20 @@ func (c *Config) fillDefaults(ctx resource.Context) error {
 	}
 
 	// Verify the namespace for any custom deployments.
-	for _, custom := range c.Custom {
-		if custom.Namespace != nil {
-			found := false
-			for _, ns := range c.Namespaces {
-				if custom.Namespace.Name() == ns.Get().Name() {
-					found = true
-					break
+	if c.Custom != nil {
+		for _, custom := range c.Custom.Get() {
+			if custom.Namespace != nil {
```

Comment:
We already have this functionality if you don't add any ns then it will automatically deploy it in all the echo ns, if we restrict it to not use any namespace, some tests will fail. Example this test uses a namespace (https://github.com/istio/istio/blob/8cb16ba82cbc044a11632d983ac9ebfb34d88112/tests/integration/security/ca_custom_root/main_test.go#L83) to add a SAN (https://github.com/istio/istio/blob/master/samples/certs/generate-workload.sh#L24). So we would require a namespace while creating the custom deployment sometimes.